Source/Filters/Video/YPVideoFiltersVC.swift

```diff
@@ -190,17 +193,39 @@ open class YPVideoFiltersVC: UIViewController, IsMediaFilterVC {
             selectTrim()
             videoView.loadVideo(inputVideo)
             videoView.pause()
-            setupGenerator()
+            setupGenerator(inputAsset)
             coverThumbSelectorView.delegate = self
-            coverThumbSelectorView.asset = inputAsset
+            if coverThumbSelectorView.asset == nil {
+                coverThumbSelectorView.asset = inputAsset
+            }
         } else {
-            if coverThumbSelectorView.asset != nil {
-                return
+            if YPConfig.video.coverSelectionTrimmed,
+               let startTime = trimmerView.startTime,
+               let endTime = trimmerView.endTime,
+               startTime != coverTrimTimes?.startTime || endTime != coverTrimTimes?.endTime {
+                let timerange = CMTimeRange(start: startTime, end: endTime)
+                mediaManager.fetchVideoUrlAndCrop(for: inputVideo.asset!, cropRect: inputVideo.cropRect!, timeRange: timerange, shouldMute: false, compressionTypeOverride: AVAssetExportPresetPassthrough) { [weak self] (url) in
+                    DispatchQueue.main.async {
+                        if let url = url {
+                            let trimmedAsset = AVAsset(url: url)
+                            self?.setupGenerator(trimmedAsset)
+                            self?.coverThumbSelectorView.asset = trimmedAsset
+                            if let coverImageTime = self?.coverImageTime, coverImageTime < startTime || coverImageTime > endTime {
+                                self?.generateCoverImageAtTime(startTime)
+                            }
+                            self?.coverTrimTimes = (startTime: startTime, endTime: endTime)
+                        } else {
+                            ypLog("YPVideoFiltersVC -> Invalid asset url.")
+
```

Comment:
Lots going on here. Might get you to talk me through this.
Also, there's a blank line here after the Log.

Reply:
What this basically does is the following:

If the new config value is on _and_ the trim value has changed since the cover photo was selected, we should re-create the asset used for the cover photo selector (and then save the trim value). 

Once the asset is created, we also check if the previous cover photo timestamp was within the new trim value and set the cover photo to be the first frame of the trimmed asset if it wasn't.